### DISCUSSION TITLE
Need help updating Configuring The Blazor Oqtane Blog Module

### DISCUSSION THREAD

**Discussion by ADefWebserver** [oqtane/oqtane.framework#856] [Q&A] [answered]
I need to update [Configuring The Blazor Oqtane Blog Module](https://blazorhelpwebsite.com/ViewBlogPost/15)

Questions:

1. Do we need to "Fix References" anymore?
2. When Oqtane is updated to .Net Core 5 will the references in Oqtane.Blogs.nuspec need to be changed from "**oqtane.framework\Oqtane.Server\bin\Debug\netcoreapp3.1\**" to "**oqtane.framework\Oqtane.Server\bin\Debug\netcoreapp5.0\**"?
3. Will the Blog be upgraded to .Net Core 5.0?
4. If yes to 3 should I do the upgrade and make the Pull Request? (if so I will need to wait for Oqtane to be upgraded)

**Comment by hishamco** [comment]
I think this should go to the Blogs module repo

**Comment by sbwalker** [comment, marked as answer]
Wow! You can definitely tell that blog post is over a year old! To answer your questions:

1. Do we need to "Fix References" anymore?
>> No. The blog module now uses standard Nuget package references - look in the csproj files
>> By convention if you place the Blog module root folder at the same level as the oqtane framework, then you will not have to modify any folder configuration in the Package project:

![image](https://user-images.githubusercontent.com/4840590/98038441-24643880-1deb-11eb-8ace-c302777fb2fb.png)

>> You may want to mention that if you compile the solution in the Debug configuration it will automatically copy the DLLs and static resources to the appropriate locations so that when you run Oqtane it will discover them automatically
>> You can also remove the section about "Build the Nuget Package Into the Oqtane Modules Folder". Instead, you would switch your configuration in Visual Studio to Release and build the solution and it will create the nuget package for you. But these instructions would probably be better at the end of the blog post where you mention "Package The Module As a NuGet Package". Compiling in Release configuration will automatically copy the nuget package to the appropriate location so that when you run Oqtane it will install it automatically.

2. When Oqtane is updated to .Net Core 5 will the references in Oqtane.Blogs.nuspec need to be changed from "**oqtane.framework\Oqtane.Server\bin\Debug\netcoreapp3.1*" to "**oqtane.framework\Oqtane.Server\bin\Debug\netcoreapp5.0*"?
>> Yes, there are a number of references which need to be changed for the module to be built on .NET 5. However since .NET 5 is not officially released yet, I would not recommend making these changes to the Blog module yet. 

3. Will the Blog be upgraded to .Net Core 5.0?
>> It should be upgraded to .NET 5 only after .NET 5 is officially released. Currently the latest .NET 5 release is RC2. This is what the Oqtane main branch has been upgraded to - but it will need to be upgraded to the official .NET 5 bits once they are released next week. At that point there will be official Oqtane 2.0 Nuget packages created based on .NET 5 and the Blog module can be upgraded to reference them.

4. If yes to 3 should I do the upgrade and make the Pull Request? (if so I will need to wait for Oqtane to be upgraded)
See the answer above.

Note that an Oqtane module which is build on .NET Core 3.1 will run just fine on Oqtane 2.0 ( built on .NET 5 ). So it would also be possible for you to update your blog post to reflect the latest changes with the Blog module without upgrading to .NET 5.

**Reply by ADefWebserver** [reply]
Thanks @sbwalker I will update the blog post this week, and then update it again when Oqtane is upgraded to .Net Core 5, then update it again if the Blog module is upgraded  to .Net Core 5.

**Reply by sbwalker** [reply]
You will not need to update the blog post when Oqtane is upgraded to .NET 5, as there is no changes required to the module - it will work seamlessly. You will only need to update the blog post when the Blog module is updated to .NET 5.

**Reply by ADefWebserver** [reply]
@sbwalker in the "\oqtane.blogs\Package\debug.cmd" it has references to: "..\..\oqtane.framework\Oqtane.Server\bin\Debug\netcoreapp3.1\". I thought those would have to be updated, no?

**Reply by sbwalker** [reply]
By convention if you place the Blog module root folder at the same level as the oqtane framework, then you will not have to modify any folder configuration in the Package project ( see screenshot above )

**Comment by ADefWebserver** [comment]
I updated the Blog post: [Configuring The Blazor Oqtane Blog Module](https://blazorhelpwebsite.com/ViewBlogPost/15)

**Reply by sbwalker** [reply]
@ADefWebserver The Oqtane 2.0 Nuget packages were published yesterday so it would be possible to upgrade the Blog module to .NET 5 now

**Reply by ADefWebserver** [reply]
> @ADefWebserver The Oqtane 2.0 Nuget packages were published yesterday so it would be possible to upgrade the Blog module to .NET 5 now

I will try to do a pull request this weekend.

**Reply by hishamco** [reply]
@ADefWebserver I pushed a PR https://github.com/oqtane/oqtane.blogs/pull/8 coz I need to do some localization updates

**Reply by ADefWebserver** [reply]
> @ADefWebserver I pushed a PR [oqtane/oqtane.blogs#8](https://github.com/oqtane/oqtane.blogs/pull/8) coz I need to do some localization updates

@hishamco - Niiiiice! Thank you!
